cpp/ql/src/Security/CWE/CWE-014/MemsetMayBeDeleted-bad.c

```diff
@@ -0,0 +1,4 @@
+char * password = malloc(PASSWORD_SIZE);
+// ... read and check password
+memset(password, 0, PASSWORD_SIZE);
+free(password);
```

Comment:
This is an unfortunate example since the query in its current state doesn't understand `free`.

Reply:
Also, this snippet was copied from the first page of https://www.usenix.org/system/files/conference/usenixsecurity17/sec17-yang.pdf. We can't risk any copyright issues, so please write a new example from or use an abbreviated form of `cpp/ql/src/experimental/Security/CWE/CWE-14/CompilerRemovalOfCodeToClearBuffers.c`.

Reply:
> This is an unfortunate example since the query in its current state doesn't understand free.

Good catch! I'll change the example to be a statically allocated buffer. That'll side-step both problems.

Implementing more model interfaces in our model of `free` sounds like a change that would be beneficial for many (future) queries.

Reply:
Fixed in 42d2a673c7023b546909cafb2e24d4ee9b425cb9.